packages/protocol/contracts/common/Accounts.sol

```diff
@@ -107,12 +117,21 @@ contract Accounts is IAccounts, Ownable, ReentrancyGuard, Initializable, UsingRe
   /**
    * @notice Setter for the wallet address for an account
    * @param walletAddress The wallet address to set for the account
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
    * @dev Wallet address can be zero. This means that the owner of the wallet
    *  does not want to be paid directly without interaction, and instead wants users to
    * contact them, using the data encryption key, and arrange a payment.
+   * @dev v, r, s constitute `signer`'s signature on `msg.sender` (unless the wallet address
+   *      is 0x0 or msg.sender).
    */
-  function setWalletAddress(address walletAddress) public {
+  function setWalletAddress(address walletAddress, uint8 v, bytes32 r, bytes32 s) public {
     require(isAccount(msg.sender), "Unknown account");
+    if (!(walletAddress == msg.sender || walletAddress == address(0x0))) {
```

Comment:
Assuming this is why, but there's nothing in the contractkit code that prevents users from calling `setWalletAddress` with a null signature but a non-zero, non-msg.sender walletAddress

Reply:
Indeed, but they would just have the TX reverted on them? It sounds like you are saying we should do a check in CK itself?

Reply:
It's kind of hard to do because we do not have access to `msg.sender` easily, so I would advocate to keep it in the responsibility of the caller?